src/components/general/buttons/MenuButton.tsx

```diff
@@ -0,0 +1,167 @@
+    const slideAnim = useState(new Animated.Value(-275))[0];
+    const navigationState = useAppSelector((state) => state.navigation);
+
+    useEffect(() => {
+        Animated.timing(slideAnim, {
+            toValue: menuVisible ? 0 : -275,
+            duration: 300,
+            useNativeDriver: true,
+        }).start();
+    }, [menuVisible]);
+
+    const toggleMenu = () => {
+        HapticFeedback(HapticForces.Light);
+        setMenuVisible((prev) => !prev); 
+    };
+
+    function isRouteActive(route: string) {
+        return navigationState.selectedNavBarRoute === route;
+    }
+
+    return (
+        <>
+            <TouchableOpacity style={styles.container} onPress={toggleMenu}>
+                <View style={styles.hamburger}>
+                    <View style={styles.bar} />
+                    <View style={styles.bar} />
+                    <View style={styles.bar} />
+                </View>
+            </TouchableOpacity>
+
+            {menuVisible && (
+                <>
+                    <TouchableOpacity style={styles.overlay} onPress={toggleMenu} />
+
+                    <Animated.View style={[styles.menuContainer, { transform: [{ translateX: slideAnim }] }]}>
+                        <Text style={styles.menuTitle}>Menu</Text>
+
+                        {navigationBarLinks.map((link) => (
+                            <TouchableOpacity
+                                key={link.route}
+                                style={styles.menuItem}
+                                onPress={() => {
+                                    HapticFeedback(HapticForces.Light);
+                                    navigation.navigate(link.route);
+                                    toggleMenu();
+                                }}
+                            >
+                                <Image source={getIcon(link.icon)} style={styles.menuIcon} />
+                                <Text style={styles.menuText}>{link.icon}</Text>
+                            </TouchableOpacity>
+                        ))}
+
+                        <SettingsButton toggleMenu={toggleMenu} />
+                    </Animated.View>
+                </>
+            )}
+        </>
+    );
+};
+
+const getIcon = (iconName: string) => {
+    switch (iconName) {
+        case 'home':
+            return require('../../../assets/images/navbarIcons/Home.png');
+        case 'articles':
+            return require('../../../assets/images/navbarIcons/Articles.png');
+        case 'quizzes':
+            return require('../../../assets/images/navbarIcons/Courses.png');
+        case 'prompts':
+            return require('../../../assets/images/navbarIcons/Prompts.png');
+        case 'WTR':
+            return require('../../../assets/images/navbarIcons/WindesheimTech.png');
+        case 'scans':
+            return require('../../../assets/images/navbarIcons/WindesheimTech.png');
+        default:
+            return null;
+    }
+};
```

Comment:
These paths are going back quite a few directories. Is there another way to import these?

Reply:
I tried some things but I couldn't get it to work, this seems to be a global thing because in other components they use the method, put this on hold for now?